.github/workflows/reusable-nox.yml

```diff
@@ -34,19 +34,27 @@ jobs:
           - session: "pip-compile"
             extra-args: "--check"
             python-versions: "3.12"
+
     name: "Run nox ${{ matrix.session }} session"
     steps:
       - name: Check out repo
         uses: actions/checkout@v5
+
       - name: Setup nox
         uses: wntrblm/nox@2025.05.01
         with:
           python-versions: "${{ matrix.python-versions }}"
+
       - name: Graft ansible-core
-        run: |
-          nox -e clone-core
-      - name: "Run nox -e ${{ matrix.session }}"
-        run: |
-          # Using GHA expression interpolation is fine here,
-          # as we control all the inputs.
-          nox -e "${{ matrix.session }}" -- ${{ matrix.extra-args }}
+        uses: ansible-community/github-action-run-nox@v1
+        with:
+          sessions: clone-core
+          force-pythons: "${{ matrix.python-versions }}"
+
+      - name: "Run nox session"
+        uses: ansible-community/github-action-run-nox@v1
+        with:
+          sessions: "${{ matrix.session }}"
+          extra-args: "${{ matrix.extra-args || '' }}"
+          force-pythons: "${{ matrix.python-versions }}"
```

Comment:
Here's examples of splitting provisioning the venv and running the logic:
* https://github.com/pyca/cryptography/blob/09fb854e009d560e5edbe6677ce53f2b8c0b6011/.github/workflows/linkcheck.yml#L39-L45
* https://github.com/ansible/receptor/blob/59e0da47a39518b9a2742c7efae7bb7985a4d9a7/.github/workflows/coverage_reporting.yml#L52-L58

TL;DR for two steps to show up separately in CI, they should be actual separate entries in the step list (for the same session). The first one would run with `nox --install-only`, and it'd be followed by `nox --no-install`.